plugins/source/azure/client/client.go

```diff
@@ -265,14 +266,31 @@ func New(ctx context.Context, logger zerolog.Logger, s *Spec) (schema.ClientMeta
 		}
 	})
 
+	oidcToken := c.pluginSpec.OIDCToken
 	var credsOptions *azidentity.DefaultAzureCredentialOptions
 	if c.Options != nil {
 		credsOptions = &azidentity.DefaultAzureCredentialOptions{ClientOptions: c.Options.ClientOptions}
 	}
 	var err error
-	c.Creds, err = azidentity.NewDefaultAzureCredential(credsOptions)
-	if err != nil {
-		return nil, err
+	if oidcToken != "" {
+		// Accessing Azure using OIDC token
+		tenantID := os.Getenv("AZURE_TENANT_ID")
```

Comment:
`AZURE_TENANT_ID` & `AZURE_CLIENT_ID` shouldn't be looked up directly by plugin.
Instead, they should also be a part of the spec.

Additionally, the logic verifying that both of these params should be added.

Reply:
@candiduslynx will add the required changes in a new PR.

Reply:
@candiduslynx Thanks